src/Symfony/Component/JsonPath/JsonPathUtils.php

```diff
@@ -159,4 +159,18 @@ private static function unescapeUnicodeSequence(string $str, int $length, int &$
 
         return mb_chr($codepoint, 'UTF-8');
     }
+
+    /**
+     * @see https://datatracker.ietf.org/doc/rfc9535/, section 2.1.1
+     */
+    public static function normalizeWhitespace(string $input): string
+    {
+        $normalized = strtr($input, [
+            "\t" => ' ',
+            "\n" => ' ',
+            "\r" => ' ',
+        ]);
+
+        return trim(preg_replace('/\s+/', ' ', $normalized));
```

Comment:
I'm wondering whether this might replace consecutive spaces inside quoted string literals, changing their meaning. Do we need to replace those consecutive spaces ? trimming will remove multiple spaces if needed.

Reply:
Ah right, I updated the method and added `testMatchFunctionWithMultipleSpacesTrimmed` to ensure we don't trim meaningful blank spaces